.circleci/config.yml

```diff
@@ -4,27 +4,34 @@ version: 2.1
 commands:
 
 jobs:
-  test_setup:
+  test:
     machine:
       image: ubuntu-2204:current
     steps:
       - checkout
       - run:
-          name: Build docker images
-          command: docker compose build
+          name: Pull Docker images
+          command: |
+            docker compose pull --quiet --ignore-buildable
+            docker pull --quiet azul/zulu-openjdk-debian:11
       - run:
-          name: Pull docker images
-          command: docker compose pull --ignore-buildable
+          name: Build Docker images
+          command: docker compose build
       - run:
           background: true
-          name: Start the compose setup
-          command: docker compose up
-      - run: ./scripts/validate-deployment.sh
+          name: Start Docker Compose in the background
+          command: docker compose up --abort-on-container-exit
       - run:
+          name: Test Docker Compose
+          command: |
+            sleep 10s
```

Comment:
Why the sleep? The test has a retry to explicitly remove the need for sleeping here.

Reply:
Docker compose is slow to start, I don't really see the benefit of trying to test it out immediately, it's like an initial delay?

Reply:
Removed in favor of fixed delay on retries